testSubSubmoduleUpdateSwitched failing locally

**Describe the bug**

On my local machine the `test_input_gitscm` was always failing with:

```
   586	======================================================================
   587	FAIL: testSubSubmoduleUpdateSwitched (test_input_gitscm.TestSubmodules.testSubSubmoduleUpdateSwitched)
   588	Test update of switched sub-submodule
   589	----------------------------------------------------------------------
   590	Traceback (most recent call last):
   591	  File "/home/ralf/projects/bob/test/unit/test_input_gitscm.py", line 836, in testSubSubmoduleUpdateSwitched
   592	    self.assertFalse(os.path.exists(os.path.join(workspace, "sub1/some/deep/path/canary.txt")))
   593	AssertionError: True is not false
```

It took me some time to find out this was due to a setting in my `~/.gitconfig`:

``` 
[submodule]    
    recurse = true
```
which I have there to automatically update the layer-submodules...

I fear this could also lead to different checkout results?

**Steps to reproduce**

```
$ git config --global submodule.recurse true
$ ./test/run-tests.sh -u test_input_gitscm
```

**Context**
- OS: Debian 12.5
- Bob version: master
- Python version: 3.11.2
- git version: 2.39.2

--- a/pym/bob/scm/git.py
+++ b/pym/bob/scm/git.py
@@ -420,7 +420,7 @@ class GitScm(Scm):
     async def __checkoutSubmodules(self, invoker):
         if not self.__submodules: return
 
-        args = ["git", "submodule", "update", "--init"]
+        args = ["git", "-c", "submodule.recurse=0", "submodule", "update", "--init"]
         if self.__shallowSubmodules:
             args += ["--depth", "1"]
         if self.__recurseSubmodules:
@@ -497,7 +497,7 @@ class GitScm(Scm):
             return {}
 
         # Sync remote URLs into our config in case they were changed
-        args = ["git", "-C", base, "submodule", "sync"]
+        args = ["git", "-c", "submodule.recurse=0", "-C", base, "submodule", "sync"]
         await invoker.checkCommand(args, cwd=self.__dir)
 
         # List all paths as per .gitmodules. This gives us the list of all
@@ -526,7 +526,7 @@ class GitScm(Scm):
             }
 
         # Do the update of safe submodules
-        args = ["git", "-C", base, "submodule", "update", "--init"]
+        args = ["git", "-c", "submodule.recurse=0", "-C", base, "submodule", "update", "--init"]
         if self.__shallowSubmodules:
             args += ["--depth", "1"]
         args.append("--")

